### DISCUSSION TITLE
About Rodrigues space

### DISCUSSION THREAD

**Discussion by yiyuxingchen** [neperfepx/neper#473] [Q&A] [answered]
@rquey 
Hello,
I know from here how to generate a Rodrigues space, https://neper.info/tutorials/orientation_color_key.html, but I still don't know how to make the following picture.
![image](https://user-images.githubusercontent.com/103731227/167876997-5e55fa10-5034-4bd4-9983-052942953bdd.png)

For example, how to generate a Rodrigues space of uniform distributions of 1000 orientations with cubic crystal symmetries.
And, what variable to output after deformation to generate Rodrigues space again.
Thanks in advance.

**Comment by xshang93** [comment]
The link is to generate a color key, not the actually distribution map you're showing here.

I might be wrong but I don't think neper can be used to generate a distribution map like the one you are showing.

**Comment by rquey** [comment, marked as answer]
(Needs Version 4.4.2-22)

Neper can indeed generate these plots, even if the process is a bit tedious for now. It will be much simpler in a future Neper version.

Here is how to proceed:

```
# generate tessellation and mesh for the polycrystal
neper -T -n 1000 -oricrysym cubic -o n1000 -reg 1
neper -M n1000.tess -for msh,sim

# generate and mesh orientation space
neper -T -n 1 -domain "rodrigues(cubic)" -o rod
neper -M rod.tess -cl 0.04 -for msh,sim

# compute odf at elements and nodes (only nodes needed)
neper -S n1000.sim -orispace rod.msh -resmesh "odf,odfn"

# visualize orientation space and color according to the ODF nodal values
neper -V n1000.sim/orispace/rod.msh    \
      -datanodecol "real:file(n1000.sim/results/mesh/odfn/odfn.step0)" \
      -datanodescale 0.2:1.8           \
      -datanodescaletitle "Orientation distribution function (m.r.d.)"  \
      -dataeltcol from_nodes           \
      -dataelt3dedgerad 0              \
      -dataelt1drad 0.002              \
      -showelt1d all                   \
      -cameracoo 4:4:3                 \
      -cameraangle 12                  \
      -cameraprojection orthographic   \
      -print img

# paste scale bar
convert +append img.png img-scalen.png img.png
```
![image](https://user-images.githubusercontent.com/2659376/174622058-bdcfb5d2-5dfa-4cab-b76d-1f05fe850d70.png)

To do the same for a deformed polycrystal, do the same with your simulation directory and use the appropriate step file.

(Note that the orientations are weighted by the volume of the grains).
